cpp/src/stream_compaction/distinct.cu

```diff
@@ -82,7 +82,12 @@ std::unique_ptr<table> distinct(table_view const& input,
   auto counting_iter = thrust::make_counting_iterator<size_type>(0);
   rmm::device_uvector<bool> index_exists_in_map(num_rows, stream, mr);
   // enumerate all indices to check if they are present in the map.
-  key_map.contains(counting_iter, counting_iter + num_rows, index_exists_in_map.begin(), hash_key);
+  key_map.contains(counting_iter,
+                   counting_iter + num_rows,
+                   index_exists_in_map.begin(),
+                   hash_key,
+                   thrust::equal_to<size_type>{},
+                   stream.value());
```

Comment:
Do we have a `get_keys` method to just extract a list of all keys from the map?

Reply:
I found `retrieve_all` but not sure what is it for?

Reply:
Good point! `retrieve_all` it is.